labs/7.md

```diff
@@ -1,90 +1,127 @@
-## Creating Certificates
-This demonstration uses MongoDB with SSL enabled. The example below will create a
-generic CA, key, and certificate. The following binaries are required.
+## Deploying *Pacman*
+Now that the Mongo cluster has been configured, it is time to deploy the *pacman* application.
+There are many different types of federated objects but they are somewhat similar to those
+non-federated objects. For more information about federated objects see the following  [examples](https://github.com/kubernetes-sigs/kubefed/tree/master/example/sample1) and
+the [user guide](https://github.com/kubernetes-sigs/kubefed/blob/master/docs/userguide.md).
 
-* [cssl](https://pkg.cfssl.org/R1.2/cfssl_linux-amd64)
-* [cfssljson](https://pkg.cfssl.org/R1.2/cfssljson_linux-amd64)
+The pacman namespace must be created and then defined as a federated namespace.
+~~~sh
+oc new-project pacman
+kubefedctl federate namespace pacman
+~~~
+
+Validate the namespace exists in the three clusters.
+~~~sh
+for i in cluster1 cluster2 cluster3; do oc get project pacman --context $i; done
+NAME     DISPLAY NAME   STATUS
+pacman                  Active
+NAME     DISPLAY NAME   STATUS
+pacman                  Active
+NAME     DISPLAY NAME   STATUS
+pacman                  Active
+~~~
+
+Modify the cluster names to reflect the names of the kubefedclusters defined above.
+~~~sh
+sed -i 's/feddemocl1/cluster1/g' ./*.yaml
+sed -i 's/feddemocl2/cluster2/g' ./*.yaml
+sed -i 's/feddemocl3/cluster3/g' ./*.yaml
+~~~
+
+For the *pacman* application, the file `pacman-federated-deployment-rs.yaml` needs to reflect the MongoDB endpoint. The MongoDB endpoint is used to save scores from the game.
+Provide the value of the MongoDB server(s) to be used for the scores to be recorded
+for the *pacman* game.
+
+### Will be patched when Mario finished AGNOSTICD
+~~~sh
+sed -i 's/replicamembershere/mongo-cluster1.apps.cluster-1.example1.com,mongo-cluster2.app.ccluster2.example1.com,mongo-cluster3.apps.cluster3.example1.com/g' 07-pacman-federated-deployment-rs.yaml
+~~~
+
+A value must be provided to be the publicly accessible address for the *pacman* application.
+~~~sh
+sed -i 's/pacmanhosthere/pacman.example1.com/g' 03-pacman-federated-ingress.yaml
+~~~
+
+Now deploy the *pacman* objects.
+~~~sh
+# Create the MongoDB secret
+oc create -n pacman -f 01-mongo-federated-secret.yaml
+# Create the service
+oc create -n pacman -f 02-pacman-federated-service.yaml
+# Create the ingress endpoint
+oc create -n pacman -f 03-pacman-federated-ingress.yaml
+# Create the pacman service serviceaccount
+oc create -n pacman -f 04-pacman-federated-service-account.yaml
+# Create the pacman cluster role
+oc create -n pacman -f 05-pacman-federated-cluster-role.yaml
+# Create the clusterrolebinding
+oc create -n pacman -f 06-pacman-federated-cluster-role-binding.yaml
+# Create the deployment
+oc create -n pacman -f 07-pacman-federated-deployment-rs.yaml
+~~~
+
+## Deploying HAProxy
```

Comment:
This could be potentially removed from the lab instructions, the HAProxy will be pre-deployed.

Reply:
Or maybe moved to a different file where we can expand information related to the labs, wdyt?

Reply:
I would say we move it once we get the agnosticd stuff in place and we can test against that